core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -279,13 +286,42 @@ public void requestConnection() {
           subchannels.get(addressIndex.getCurrentAddress()).getState();
       if (subchannelState == IDLE) {
         subchannel.requestConnection();
+        scheduleConnection();
       } else if (subchannelState == CONNECTING || subchannelState == TRANSIENT_FAILURE) {
         addressIndex.increment();
         requestConnection();
       }
     }
   }
 
+
+  /**
+  * Happy Eyeballs
+  * Schedules connection attempt to happen after a delay to the next available address.
+  */
+  private void scheduleConnection() {
```

Comment:
Do we need a `throwIfNotInThisSynchronizationContext()` for this and other similar methods?

Reply:
For here, you only need them if it'd make you more confident this code calls things on the correct thread. There doesn't seem to be much risk here, so I wouldn't expect it.

We had some other APIs it would be easy to accidentally call at the wrong time, and some that we changed from thread-safe to only-call-from-sync-context. That's where you probably saw the throws today.